### PR TITLE
Updates python packages to their latest version

### PR DESCRIPTION
**Pull request by smuzaffar**
This PR updates all the python packages to use their latest version.

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test for slc7_ppc64le_gcc9

**Comment by smuzaffar** [comment]
please test for slc7_aarch64_gcc9

**Comment by smuzaffar** [comment]
please test for cc8_amd64_gcc9

**Comment by smuzaffar** [comment]
please test for slc7_amd64_gcc11

**Comment by smuzaffar** [comment]
test parameters:
- full_cmssw = true

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

File: build-with-pip.file [hunk 1]
```diff
@@ -43,6 +43,9 @@
 
 %define pip pip3&package_dependency=py3-pip
 Requires: python3 py3-pip py3-setuptools
+%if "%{n}" != "py3-wheel"
```
Comment (davidlange6) [inline build-with-pip.file:46]: hi @smuzaffar - I was looking at python packages too.. Getting errors related to wheels  is likely a sign that you need to update setup tools

Comment (smuzaffar) [reply build-with-pip.file:46]: I tried building latest setuptools but was not able to build it itself. It was complainging about missing build/script-39 directory.

**Comment by davidlange6** [comment]
Interesting ... I found a workaround for that but then hit other issues.. Will investigate later.

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test for slc7_aarch64_gcc9

**Comment by smuzaffar** [comment]
please test for slc7_ppc64le_gcc9

**Comment by smuzaffar** [comment]
please test for cc8_amd64_gcc9

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test for cc8_amd64_gcc9

**Comment by smuzaffar** [comment]
please test for cc8_amd64_gcc9

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
looks good

**Comment by smuzaffar** [comment]
+externals

**Comment by Dr15Jones** [comment]
Are all the failed comparisons because the histogram entries for this PR all missing?

**Comment by smuzaffar** [comment]
@Dr15Jones , for production arch (slc7_amd64_gcc900), there are no comparison diffs ( https://github.com/cms-sw/cmsdist/pull/7263#issuecomment-913391021 ).  
For CC8 ( https://github.com/cms-sw/cmsdist/pull/7263#issuecomment-913569935 ) , yes  there are a  lot of diffs but I do not know why (may be due to slc7 baseline vs cc8 PR)

**Pull request merged**